### ISSUE TITLE
The text field is not auto focus when the console show

### ISSUE THREAD

**Issue by fran6k** [BarionLP/UnityConsole#5]
this project is awesome, but some small point problem me.

after console show, i must press any key so that the text field focus, then i can input something.

can we make it auto focus when i open the console?

**Comment by BarionLP** [comment]
I tried to do it but i somehow couldn't get it working. If you can provide some help it would be great

**Comment by fran6k** [comment]
i made it work.

first,based on my experience. i guess when i type something, unity need switch input system.
then i find this article.
 [https://docs.unity.cn/cn/2022.3/Manual/UIE-Runtime-Event-System.html](url) \
ui toolkit with new input system need some addtional setting. we need to disable old input manager to prevent some "detect".

secondly, use InputElement.Focus() method to focus. use IEnumerator to excute focus in next update frame time. because in this update frame time the ui element is not visible.

```csharp
  IEnumerator focusDelay()
        {
            yield return new WaitForSeconds(0.1f);
            InputElement.Focus();
        }
```

**Comment by BarionLP** [comment]
> secondly, use InputElement.Focus() method to focus. use IEnumerator to excute focus in next update frame time. because in this update frame time the ui element is not visible.

thanks! I'll fix it today

**Comment by BarionLP** [comment]
do you need it to work in 2022.3? 
my current implementation uses `Awaitable` which got introduced in 2023.1

**Comment by BarionLP** [comment]
I implemented a fallback for 2022.3 as it is still LTS but i couldn't test it

**Comment by fran6k** [comment]
> do you need it to work in 2022.3? my current implementation uses `Awaitable` which got introduced in 2023.1

i guess nop, higher version is ok
